lwgenerate/azure/azure.go

```diff
@@ -0,0 +1,458 @@
+package azure
+
+import (
+	"github.com/hashicorp/hcl/v2/hclwrite"
+	"github.com/lacework/go-sdk/lwgenerate"
+	"github.com/pkg/errors"
+)
+
+type GenerateAzureTfConfigurationArgs struct {
+	// Should we configure Activity Log integration in LW?
+	ActivityLog bool
+
+	// Should we add Config integration in LW?
+	Config bool
+
+	// Should we create an Active Directory integration
+	CreateAdIntegration bool
+
+	// If Config is true, give the user the opportunity to name their integration. Defaults to "TF Config"
+	ConfigIntegrationName string
+
+	// If ActivityLog is true, give the user the opportunity to name their integration. Defaults to "TF activity log"
+	ActivityLogIntegrationName string
+
+	// Active Directory application Id
+	AdApplicationId string
+
+	// Active Directory password
+	AdApplicationPassword string
+
+	// Active Directory Enterprise app object id
+	AdServicePrincipalId string
+
+	// Should we use the management group, rather than subscription
+	ManagementGroup bool
+
+	// Management Group ID to set
+	ManagementGroupId string
+
+	// List of subscription Ids
+	SubscriptionIds []string
+
+	// Grant read access to ALL subscriptions
+	AllSubscriptions bool
+
+	// Storage Account name
+	StorageAccountName string
+
+	// Storage Account Resource Group
+	StorageAccountResourceGroup string
+
+	// Should we use existing storage account
+	ExistingStorageAccount bool
+
+	// Azure region where the storage account for logging resides
+	Location string
+}
+
+// Ensure all combinations of inputs are valid for supported spec
+func (args *GenerateAzureTfConfigurationArgs) validate() error {
+	// Validate one of config or activity log was enabled; otherwise error out
+	if !args.ActivityLog && !args.Config {
+		return errors.New("audit log or config integration must be enabled")
+	}
+
+	// Validate that active directory settings are correct
+	if !args.CreateAdIntegration && (args.AdApplicationId == "" || args.AdServicePrincipalId == "" || args.AdApplicationPassword == "") {
```

Comment:
I'm wondering whether this arg should actually be inverted and be `useExistingAdApplication` ?
That aligns more closely with the TF module vars.

Reply:
I see further down in the code why this is named this way. We are setting the TF var `use_existing_ad_application` to true in each scenario.

Reply:
I am setting that to true in each occasion 
1) When we create the AD application, using the returned terraform attributes
2) When we supply details for an already existing AD integration

I was under the impression that there must be an AD for this to work

Reply:
So that's not strictly true, but this relates to the first long comment that I left on this PR a while back. We have made the decision to show the Azure AD module at the top level. In reality, we can do the AD creation within the Config and/or the Activity log integration, similar to what we do in GCP.

Based on the decision made previously, I think what you have implemented here is correct, it just took me a bit to get the full picture...